File: XKS.Desktop/View/MainView.cs
using Gtk;

namespace XKS.View
{
	public sealed class MainView
	{
		public Button        NewItemButton         { get; }
		public Button        BackButton            { get; }
		public StackSwitcher DeckModeStackSwitcher { get; }
		public Stack         MainStack             { get; }
		public Viewport      DeckListViewport      { get; }
		public Stack         DeckModeStack         { get; }
		public Box           QuestionViewBox       { get; }
		public Paned         EditStack             { get; }

		public MainView(Button        newItemButton,
		                Button        backButton,
		                StackSwitcher deckModeStackSwitcher,
		                Stack         mainStack,
		                Viewport      deckListViewport,
		                Stack         deckModeStack,
		                Box           questionViewBox,
		                Paned         editStack)
		{
			NewItemButton = newItemButton;
			BackButton = backButton;
			DeckModeStackSwitcher = deckModeStackSwitcher;
			MainStack = mainStack;
			DeckListViewport = deckListViewport;
			DeckModeStack = deckModeStack;
			QuestionViewBox = questionViewBox;
			EditStack = editStack;
		}
	}
}

File: cpp/xks/view/MainView.cpp
/**
 * Copyright 2020 Vitalijus Dobrovolskis
 *
 * Created by Vitalijus Dobrovolskis on 06.02.2020
 */

#include "MainView.h"

xks::view::MainView::MainView(const RefPtr<Builder>& builder) : MainView() {
	builder->get_widget("deckListViewport", deck_list_viewport);
	builder->get_widget("mainStack", main_stack);
	builder->get_widget("backButton", back_button);
	builder->get_widget("newItemButton", new_item_button);

	builder->get_widget("deckModeStackSwitcher", deck_mode_stack_switcher);
	builder->get_widget("deckModeStack", deck_mode_stack);
}

Button& xks::view::MainView::get_back_button() const {
	return *back_button;
}

Button& xks::view::MainView::get_new_item_button() const {
	return *new_item_button;
}

Stack& xks::view::MainView::get_main_stack() const {
	return *main_stack;
}

StackSwitcher& xks::view::MainView::get_deck_mode_stack_switcher() const {
	return *deck_mode_stack_switcher;
}

Stack& xks::view::MainView::get_deck_mode_stack() const {
	return *deck_mode_stack;
}

Viewport& xks::view::MainView::get_deck_list_viewport() const {
	return *deck_list_viewport;
}